### ISSUE TITLE
space-{after,before}-type-colon not working with regular functions

### ISSUE THREAD

**Issue by renke** [gajus/eslint-plugin-flowtype#8]
given the code

```js
export default function(a: Object, b: Object): Object {
  return a + b;
}

export const foo = (a: Object, b: Object): Object => {
  return a + b;
};
```

and the config

```js
{
  "flowtype/space-after-type-colon": [2, "always"],
  "flowtype/space-before-type-colon": [2, "always"],
}
```

I only get these errors

```sh
5:21  error  There must be a space before "a" parameter type annotation colon  flowtype/space-before-type-colon
5:32  error  There must be a space before "b" parameter type annotation colon  flowtype/space-before-type-colon
```

I would have expected to get these errors for the default exported function, too.

**Comment by gajus** [comment]
@renke I cannot think of a reason. As you can see, test for the cases you have used in the example pass.

https://github.com/gajus/eslint-plugin-flowtype/commit/db6c6db64ff3b2c7523c5ea3add6d2a49f19fdff

**Comment by gajus** [comment]
Share your `.babelrc.` please. I am guessing it could have to do with order or plugin/ preset declaration.

**Comment by renke** [comment]
This is `.babelrc`

```json
{
  "presets": [
    "stage-0",
    "es2015",
    "react",
  ],
  "plugins": [
    "add-module-exports",
    "transform-decorators-legacy",
    "transform-runtime"
  ]
}
```

I tried it without the plugins and permuted presets, but it still didn't work.

Relevant package versions:
- eslint 2.4.0
- babel-eslint 6.0.0-beta.6
- eslint-plugin-flowtype 2.2.0

**Comment by gajus** [comment]
Looks like a bug in `babel-eslint`.

```js
export default function (foo: string) {}
```

If you inspect `foo` `Identifier` node, it is:

```
Node {
  type: 'Identifier',
  start: 25,
  end: 36,
  loc:
   SourceLocation {
     start: Position { line: 1, column: 25 },
     end: Position { line: 1, column: 36 } },
  name: 'foo',
  typeAnnotation:
   Node {
     type: 'TypeAnnotation',
     start: 28,
     end: 36,
     loc: SourceLocation { start: [Object], end: [Object] },
     typeAnnotation:
      Node {
        type: 'StringTypeAnnotation',
        start: 30,
        end: 36,
        loc: [Object],
        range: [Object],
        _babelType: 'StringTypeAnnotation' },
     range: [ 28, 36 ],
     _babelType: 'TypeAnnotation' },
  range: [ 25, 36 ],
  _babelType: 'Identifier' }
```

Notice that the `node.end` is 36, i.e. it considers that `Identifier` is `foo: string`.

This is inconsistent with the way `ArrowFunctionExpression` is interpreted, e.g.

```js
export default (foo: string) => {}
```

Here `Identifier` node is:

```
Node {
  type: 'Identifier',
  start: 16,
  end: 19,
  loc:
   SourceLocation {
     start: Position { line: 1, column: 16 },
     end: Position { line: 1, column: 19 } },
  name: 'foo',
  typeAnnotation:
   Node {
     type: 'TypeAnnotation',
     start: 19,
     end: 27,
     loc: SourceLocation { start: [Object], end: [Object] },
     typeAnnotation:
      Node {
        type: 'StringTypeAnnotation',
        start: 21,
        end: 27,
        loc: [Object],
        range: [Object],
        _babelType: 'StringTypeAnnotation' },
     range: [ 19, 27 ],
     _babelType: 'TypeAnnotation' },
  range: [ 16, 19 ],
  _babelType: 'Identifier' }
```

Here `node.start` and `node.end` capture only `foo`.

I will raise an issue with `babel-eslint`.

**Comment by gajus** [comment]
Not going to patch this in `eslint-plugin-flowtype`. Will wait until `babel-eslint` is patched.

**Comment by renke** [comment]
Thanks for your effort. I will track the issue in babel-eslint and wait for a new release.

**Comment by danharper** [comment]
This is now fixed as of #47.
